src/types/durableClientTypes.ts

```diff
@@ -1,9 +1,226 @@
-import { FunctionInput } from "@azure/functions";
+import { FunctionInput, HttpRequest, HttpResponse } from "@azure/functions";
+import { EntityId, EntityStateResponse, HttpManagementPayload, IHttpResponse } from "../classes";
+import { DurableOrchestrationStatus } from "../durableorchestrationstatus";
+import { IHttpRequest } from "../ihttprequest";
+import { OrchestrationRuntimeStatus } from "../orchestrationruntimestatus";
+import { PurgeHistoryResult } from "../purgehistoryresult";
 
 export interface DurableClientInput extends FunctionInput {
     type: "durableClient";
 }
 
+/**
+ * Client for starting, querying, terminating and raising events to
+ * orchestration instances.
+ */
+export interface DurableClient {
```

Comment:
> ```
> // before
> const client = df.getClient(context)
> // client. -> no intellisense
> ```

This should still give intellisense. You don't need `: DurableOrchestrationClient` to get intellisense because the language service is smart enough to figure out the type of `client` based on the return type of `getClient`.

I think there's a few breaking changes here:
1. Changing the name from `DurableOrchestrationClient` to `DurableClient`
2. Changing the return of `getClient` from a class to an interface (affecting things like `instanceof`)

With all breaking changes that we're doing, could we get some reasoning on why these are worth it? If we're not sure, let's err on the side of maintaining the old behavior.

Reply:
Does this still count as a breaking change when users couldn't access the `DurableOrchestrationClient` class from the root of the package? Before, to do something like `instance of`, you had to import it like this:

```TS
import { DurableOrchestrationClient } from 'durable-functions/lib/src/durableorchestrationclient';

if (client instanceof DurableOrchestrationClient)  {
     // do something
}
```

Regardless, here's how I would justify these changes:
1. The name "orchestration client" is outdated, since now durable clients can interact with both entities and orchestrations. This is why the name of the input binding type was also updated from `orchestrationClient` to `durableClient`. There's also an issue #154 to update the name of this class from `DurableOrchestrationClient` to `DurableClient`, so now is a good chance to do that :)
2. I feel less strongly about this now, but I just figured this offers cleaner separation between API declarations and implementation.

Reply:
> Does this still count as a breaking change when users couldn't access the DurableOrchestrationClient class from the root of the package? Before, to do something like instance of, you had to import it like this:

Yes. The code worked before and will break with these changes - that's about as simple as it gets. There are rare exceptions which we could discuss offline if you want 

1. I'm fine with the change given that reasoning, but we should handle it properly. I'm glad we already have an issue - please label it "v3"/"breaking", link it to this PR in the description, and make sure it gets closed when this PR is merged.
2. I'm fine with moving it to a different file to separate API declarations from implementation. You can keep it as a class though. Instead of `export interface` just do `export declare class`

Reply:
My 2 cents:
On (1): Yes I agree we should support the name `DurableClient`. However, [this](https://github.com/Azure/azure-functions-durable-js/issues/154) ticket asks for it to be implemented in a non-breaking way so I will have to ask for the same. Can we use the good old trick of exposing the new name while keeping the old one with a `@deprecated` flag?

(2) +1 to keep it as a class.

Reply:
On 1. @ejizba I've done as you requested, but I will also add that this issue was filed before the programming model and this new v3.x release was a thing, so I'm not sure if we should still keep the issue to consider doing this (in a non-breaking way) in v2.x too. Relatedly, @davidmrdavid I'd say maybe we should do this in a non-way breaking in v2.x if we want to, but I'd rather not keep the old name around for this release since it wasn't even exported from the root of the package and it wasn't in any of our docs. It's _technically_ a breaking change, but I don't personally feel bad about breaking this one.

On 2. done. I've converted this to a class declaration instead of an interface.

Reply:
@hossam-nasr: Your reasoning for (1) makes sense to me. I'm ok with using the new name for v3 and having backwards-compatibility in v2, as you suggested.